rich/live.py

```diff
@@ -0,0 +1,347 @@
+import sys
+from threading import Event, RLock, Thread
+from typing import IO, Any, List, Optional
+
+from .__init__ import get_console
+from .console import Console, ConsoleRenderable, RenderableType, RenderHook
+from .control import Control
+from .jupyter import JupyterMixin
+from .live_render import LiveRender
+from .progress import _FileProxy
+from .segment import Segment
+
+
+class _RefreshThread(Thread):
+    """A thread that calls refresh() at regular intervals."""
+
+    def __init__(self, live: "Live", refresh_per_second: float) -> None:
+        self.live = live
+        self.refresh_per_second = refresh_per_second
+        self.done = Event()
+        super().__init__()
+
+    def stop(self) -> None:
+        self.done.set()
+
+    def run(self) -> None:
+        while not self.done.wait(1 / self.refresh_per_second):
+            self.live.refresh()
+
+
+class Live(JupyterMixin, RenderHook):
+    def __init__(
+        self,
+        renderable: RenderableType = "",
+        *,
+        console: Console = None,
+        auto_refresh: bool = True,
+        refresh_per_second: float = 4,
+        transient: bool = False,
+        redirect_stdout: bool = True,
+        redirect_stderr: bool = True,
+        hide_overflow: bool = True,
+    ) -> None:
+        """Renders an auto-updating live display of any given renderable.
```

Comment:
For some reason Sphinx won't pick up the docstring from __init__. You need to move it up a level to the class docstring.

Reply:
ttps://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autoclass_content Is the reason it only gets it from the class. 

I will move it to the class.

Reply:
TIL!